server/market/market.go

```diff
@@ -587,6 +588,163 @@ func (m *Market) SubmitOrder(rec *orderRecord) error {
+// 2. Unlocks the order coins.
+// 3. Updates the storage with the new cancel order and cancels the existing limit order.
+// 3. Responds to the client that the order was received.
+// 4. Sends the unbooked order to the order feeds.
+// 5. Creates a match object, stores it, and notifies the client of the match.
+func (m *Market) processCancelOrderWhileSuspended(rec *orderRecord, errChan chan<- error) {
+	co, ok := rec.order.(*order.CancelOrder)
+	if !ok {
+		errChan <- ErrInvalidOrder
+		return
+	}
+
+	if cancelable, err := m.CancelableBy(co.TargetOrderID, co.AccountID); !cancelable {
+		errChan <- err
+		return
+	}
+
+	m.bookMtx.Lock()
+	delete(m.settling, co.TargetOrderID)
+	lo, ok := m.book.Remove(co.TargetOrderID)
+	m.bookMtx.Unlock()
+	if !ok {
+		errChan <- ErrTargetNotCancelable
+		return
+	}
+
+	m.unlockOrderCoins(lo)
+
+	sTime := time.Now().Truncate(time.Millisecond).UTC()
+	co.SetTime(sTime)
+
+	// Create the client response here, but don't send it until the order has been
+	// committed to the storage.
+	respMsg, err := m.orderResponse(rec)
+	if err != nil {
+		errChan <- fmt.Errorf("Failed to create order response: %v", err)
+		return
+	}
+
+	dur := int64(m.EpochDuration())
+	now := encode.UnixMilli(time.Now())
+	epochIdx := now / dur
+	if err := m.storage.NewArchivedCancel(co, epochIdx, int64(m.EpochDuration())); err != nil {
+		errChan <- err
+		return
+	}
+	if err := m.storage.CancelOrder(lo); err != nil {
+		errChan <- err
+		return
+	}
+
+	// If we fail to send the response, then we send this error at the end of
+	// the function. If we send the error here and return the function, then
+	// the match will not be recorded and the server DB will be in an inconsistent
+	// state.
+	sendResponseErr := m.auth.Send(rec.order.User(), respMsg)
+	if sendResponseErr != nil {
+		log.Errorf("Failed to send cancel order response: %v", err)
+	}
+
+	sig := &updateSignal{
+		action: unbookAction,
+		data: sigDataUnbookedOrder{
+			order:    lo,
+			epochIdx: 0,
+		},
+	}
+	m.sendToFeeds(sig)
+
+	match := order.Match{
+		Taker:    co,
+		Maker:    lo,
+		Quantity: lo.Remaining(),
+		Rate:     lo.Rate,
+		Epoch: order.EpochID{
+			Idx: uint64(epochIdx),
+			Dur: m.EpochDuration(),
+		},
+		FeeRateBase:  m.getFeeRate(m.Base(), m.baseFeeFetcher),
+		FeeRateQuote: m.getFeeRate(m.Quote(), m.quoteFeeFetcher),
+	}
+	m.storage.InsertMatch(&match)
+	makerMsg, takerMsg := matchNotifications(&match)
+	m.auth.Sign(makerMsg)
+	m.auth.Sign(takerMsg)
+	msgs := []msgjson.Signable{makerMsg, takerMsg}
+	req, err := msgjson.NewRequest(comms.NextID(), msgjson.MatchRoute, msgs)
+	if err != nil {
+		log.Errorf("Failed to create match reqeust: %v", err)
+	} else {
+		err = m.auth.Request(rec.order.User(), req, func(_ comms.Link, resp *msgjson.Message) {
+			m.processMatchAcksForCancel(rec.order.User(), resp)
+		})
+		if err != nil {
+			log.Errorf("Failed to send match reqeust: %v", err)
+		}
+	}
+
+	errChan <- sendResponseErr
```

Comment:
I think that sending `sendResponseErr` back is not super valuable, at least not any more than errors from `InsertMatch` or `auth.Request`.  I do agree with your comment way above that even if a send fails, we still need to do `InsertMatch` though.
Any errors will find their way back to `handleCancel` and then up to the comms message handling, which generally just logs since that's all there is to do.
In short, I think it's fine to log send/request errors in this function, but be sure to return errors from "state" related issues like an error from `InsertMatch`.  That might be different if `processMatchAcksForCancel` actually had some effect, but the only error of real consequence is the one from `InsertMatch`.

Reply:
Returning that error makes much more sense. I've updated it.